clang/lib/Sema/SemaChecking.cpp

```diff
@@ -9827,6 +9827,9 @@ static IntRange GetExprRange(ASTContext &C, const Expr *E, unsigned MaxWidth,
     return IntRange(BitField->getBitWidthValue(C),
                     BitField->getType()->isUnsignedIntegerOrEnumerationType());
 
+  if (GetExprType(E)->isVoidType())
+    return IntRange{0, true};
```

Comment:
I think these changes are fine as-is, but I do wonder if a cleaner model for us to aim for is to make the return value `std::optional` and return no valid object in this case; a void expression has *no* values it can take, and that's different than a value range of [0,0].

Reply:
@AaronBallman thanks for the feedback. Do you mean the [following changes](https://github.com/llvm/llvm-project/pull/112081/commits/9c2a745ed365449be45cd062f29c89cabde0f514)?

Reply:
@AaronBallman thanks for the feedback. Do you mean the [following changes](https://github.com/llvm/llvm-project/pull/112081/commits/9c2a745ed365449be45cd062f29c89cabde0f514)?

Reply:
@AaronBallman thanks for the feedback. Do you mean the [following changes](https://github.com/llvm/llvm-project/pull/112081/commits/9c2a745ed365449be45cd062f29c89cabde0f514)?

Reply:
@AaronBallman @shafik if these changes look good to you, would you mind proceeding with the merge? thanks